### PR TITLE
Update version.rs segfault and WithAutoRefresh rename

### PR DESCRIPTION
**Pull request by zvonkok**
~~Initialize Cache with auto_refresh true according to CDI go-impl~~
Rename WithAutoRefresh and adjust to CdiOption. 
Updated version.rs to not panic

### CODE REVIEW

**Comment by zvonkok** [comment]
@ChengyuZhu6 @Apokleos PTAL

**Comment by zvonkok** [comment]
Fixes: #40

File: src/version.rs [hunk 1]
```diff
@@ -165,7 +165,7 @@ fn requires_v040(spec: &CDISpec) -> bool {
     spec.devices
         .iter()
         .map(|d| &d.container_edits)
-        .chain(std::iter::once(spec.container_edits.as_ref().unwrap()))
+        .chain(spec.container_edits.as_ref().into_iter())
```
Comment (Apokleos) [inline src/version.rs]: Yes, great!

File: src/cache.rs [hunk 2]
```diff
@@ -79,17 +77,15 @@ pub struct Cache {
...
-            specs: HashMap::new(),
-            devices: HashMap::new(),
-            errors: HashMap::new(),
-            dir_errors: HashMap::new(),
-            auto_refresh: true,
-        }
-    ));
+pub fn new_cache(options: Vec<CdiOption>) -> Arc<Mutex<Cache>> {
+    let cache = Arc::new(Mutex::new(Cache {
+        spec_dirs: Vec::new(),
```
Comment (Apokleos) [inline src/cache.rs]: What's about adding a 'new' method with parameter `auto_refresh: bool` for `struct Cache`?

**Review by Apokleos — COMMENTED** [review]
Thx @zvonkok 
Just one comment left to discuss introducing `Cache::new(auto_refresh:bool) {}`

File: src/cache.rs (hunk 2 above)
Comment (zvonkok) [reply src/cache.rs]: What is the purpose of having an additional "new_cache" method?

Comment (zvonkok) [reply src/cache.rs]: Wait wrong thinking we do not need this. We're setting with_auto_refresh to true in the default cache, reverting this change!!

Comment (zvonkok) [reply src/cache.rs]: Removed that, resolved.

**Comment by zvonkok** [comment]
@Apokleos Also fixed a bug with edits, what we missed is that we might only have dev.edits and if spec.edits is None it will skip adding the dev.edits here is an example
```
{
            "cdiVersion": "0.6.0",
            "kind": "kata.com/gpu",
            "devices": [
                {
                "name": "0",
                "annotations": {
                    "whatever": "false",
                    "whenever": "true"
                },
                "containerEdits": {
                    "deviceNodes": [
                    {
                        "path": "/dev/zero"
                    }
                    ]
                }
                }
            ]
        }
```

**Review by ChengyuZhu6 — APPROVED** [review]
LGTM

File: src/cache.rs (hunk 2 above)
Comment (Apokleos) [reply src/cache.rs]: > What is the purpose of having an additional "new_cache" method?
>
Yeah, I notice that [Cache {}](https://github.com/cncf-tags/container-device-interface-rs/pull/39/commits/3a01b8c20b22db50d3aa8de511ef1c60f2f2ebb4) is also introduced, and if we only use the mode of `auto_refresh=true` in our plan ?

**Review by Apokleos — APPROVED** [review]
I think we should first fix related issues and make it merged. LGTM!

**Pull request merged**